src/farcaster/getFarcasterUserAddress.ts

```diff
@@ -0,0 +1,48 @@
+import { getCustodyAddressForFidNeynar } from '../utils/neynar/user/getCustodyAddressForFidNeynar';
+import { getVerifiedAddressesForFidNeynar } from '../utils/neynar/user/getVerifiedAddressesForFidNeynar';
+import { GetFarcasterUserAddressResponse } from './types';
+
+type GetFarcasterUserAddressOptions =
+  | {
+      neynarApiKey?: string; // default to onchain-kit's default key
+      hasCustodyAddresses?: boolean; // default to true
+      hasVerifiedAddresses?: boolean; // default to true
+    }
+  | undefined;
+
+/**
+ * Get the user address for a given fid
+ * @param fid  The farcaster id
+ * @param GetFarcasterUserAddressOptions  The options to specify the type of addresses to get and the neynar api key
+ * @returns the custory address and/or verified addresses. If there is an error, it returns null
+ */
+async function getFarcasterUserAddress(
+  fid: number,
+  options?: GetFarcasterUserAddressOptions,
+): Promise<GetFarcasterUserAddressResponse | null> {
+  try {
```

Comment:
@Sneh1999 side not for the next PR, large try/catch are a bad practice in Javascript, as each Try/Catch will hold on memory almost everything inside a `try` till the end of the execution. It's advice to have smaller try/catch.